src/Discord.Net.Commands/Info/CommandInfo.cs

```diff
@@ -68,18 +68,48 @@ internal CommandInfo(CommandBuilder builder, ModuleInfo module, CommandService s
         {
             services = services ?? EmptyServiceProvider.Instance;
 
-            foreach (PreconditionAttribute precondition in Module.Preconditions)
+            foreach (IGrouping<int, PreconditionAttribute> preconditionGroup in Module.Preconditions.GroupBy(p => p.Group))
```

Comment:
I think refactoring the body of these foreaches into a local method would be best for maintainability and deduplicating the code